core/spec/models/spree/order/state_machine_spec.rb

```diff
@@ -113,9 +113,18 @@
         end
       end
 
-      it "should automatically refund all payments" do
+      context "with a pending payment" do
+        let!(:payment) { create(:payment, order: order, amount: order.total, state: "pending") }
+
+        it "should change the payment state whe the payment is not completed" do
+          expect(order).to be_allow_cancel
+          expect { order.cancel! }.to change{ payment.reload.state }.to("void")
+        end
+      end
+
+      it "should not change the payment state whe the payment is completed" do
         expect(order).to be_allow_cancel
```

Comment:
Shouldn't this expectation fail now?

Reply:
Mhh, no. The completed payment state doesn't change. It stays `completed` but a refund will be issued.

Reply:
Anyway, there's a typo in the spec description which I'm going to fix.